CMakeLists.txt

```diff
@@ -26,10 +26,16 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
     set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release")
 endif()
 
+# This block, particularly the versions used, should be kept in sync with
+# samples/minimal/CMakeLists.txt.
```

Comment:
Could we have some `apple_target.cmake` and include it from both places?

Reply:
Likely, but earlier I decided not to look into that for dependency reasons mentioned in the commit message. For that same reason I didn't, from a [user's POV](https://github.com/wxWidgets/wxWidgets/blob/c069a62bd6e2a6fa6edc84a12b5fb466cd3b0386/samples/minimal/CMakeLists.txt#L11), pollute minimal's CMakeLists.txt by repeating the "keep in sync" message there as well.

The sample includes the wxWidgets use file but that file lives outside of wx and any changes would always require the latest CMake so that doesn't seem like an option.

There's an `OPTIONAL` keyword for includes, this works from samples/minimal/CMakeLists.txt:

`include(apple_target.cmake OPTIONAL)`

I think that's the least intrusive (when not wanting to keep two pieces of code in sync) but perhaps it's better to just have the file under build/cmake/ , despite a user seeing that path then.

Of course it's also possible to force inclusion as usual and risk errors (which is probably annoying if one is targeting non-Apple), it just wouldn't be my preference as not setting the target OS shouldn't by default result in a problem for macOS and iOS (right now it is for the latter due to arch differences but that's another issue).

Reply:
OK, I didn't think about it, but you're right, this is not ideal for the users, so let's keep things simple and live with this small amount of duplicated code. Sorry for not thinking it through before making this comment.

Reply:
No problem at all and thanks for reviewing despite, like me, not knowing about CMake (my knowledge doesn't go much further then pressing the... Generate button once again in case of errors).
No changes are needed (even if minimal iOS may have to change it can be done later on), but I'll wait a few more days to see if Maarten has something to say just in case.